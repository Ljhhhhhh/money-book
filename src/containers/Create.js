import React from 'react'

const Create = ({match}) => {
  return <h1>this is create {match.params.id} page</h1>
}

export default Create