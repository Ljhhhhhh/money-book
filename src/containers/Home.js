import React, { Component, Fragment, useState, useCallback } from "react";
import PriceList from "../components/PriceList";
import ViewTab from "../components/ViewTab";
import MonthPicker from "../components/MonthPicker";
import CreateBtn from "../components/CreateBtn";
import TotalPrice from "../components/TotalPrice";
import logo from "../logo.svg";
import { LIST_VIEW, CHART_VIEW, TYPE_INCOME, TYPE_OUTCOME, parseToYearAndMonth } from "../utility";
const categorys = {
  "1": {
    id: 1,
    name: "旅游",
    type: "outcome",
    iconName: "ios-plane"
  },
  "2": {
    id: 2,
    name: "旅游",
    type: "outcome",
    iconName: "ios-plane"
  }
}
const items = [
  {
    id: 1,
    title: "去旅游",
    price: 2000,
    date: "2019-5-1",
    cid: 1
  },
  {
    id: 2,
    title: "去吃大餐",
    price: 400,
    date: "2019-5-2",
    cid: 2
  }
];

class Home extends Component {
  constructor(props) {
    super(props)
    this.state = {
      items,
      currentDate: parseToYearAndMonth(),
      tabView: LIST_VIEW
    }
  }

  changeView = () => {
    
  }

  changeDate = () => {

  }
  modifyItem = () => {

  }

  createItem = () => {

  }

  deleteItem = () => {

  }

  render() {
    const {items, currentDate, tabView} = this.state;
    const itemsWithCategory = items.map(item => {
      item.category = categorys[item.cid]
      return item
    })
    let totalIncome = 0,
      totalOutcome = 0;
    items.forEach(item => {
      if (item.category.type === TYPE_OUTCOME) {
        totalOutcome += item.price;
      } else {
        totalIncome += item.price;
      }
    });
    return (
      <Fragment>
        <header className="app-header">
          <div className="row mb-5">
            <img src={logo} className="app-logo" alt="" />
          </div>
          <div className="row">
            <div className="col">
              <MonthPicker year={currentDate.year} month={currentDate.month} onChange={this.changeDate} />
            </div>
            <div className="col">
              <TotalPrice income={totalIncome} outcome={totalOutcome} />
            </div>
          </div>
        </header>
        <div className="content-area py3 px3">
          <ViewTab activeTab={tabView} onTabChange={this.changeView} />
          <CreateBtn onClick={this.createItem} />
          <PriceList items={itemsWithCategory} onModifyItem={this.modifyItem} onDeleteItem={this.deleteItem} />
        </div>
      </Fragment>
    );
  }
}

export default Home;
