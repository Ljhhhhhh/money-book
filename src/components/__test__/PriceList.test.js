import React from "react";
import {shallow} from 'enzyme'
import Ionicon from 'react-ionicons'
import PriceList from "../PriceList";
import {items, categorys} from '../../containers/Home'

const itemsWithCategory = items.map(item => {
  item.category = categorys[item.cid]
  return item
})

const props = {
  items: itemsWithCategory,
  onModifyItem: jest.fn(),
  onDeleteItem: jest.fn()
}

let wrapper
describe('test pricelist component', () => {
  beforeEach(() => {
    wrapper = shallow(<PriceList {...props} />)
  })
  it('should render the compenent to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correct price items length', () => {
    expect(wrapper.find('.list-group-item').length).toEqual(itemsWithCategory.length)
  })
  it('should render correct icon and price for each item', () => {
    const iconList = wrapper.find('.list-group-item').first().find(Ionicon)
    expect(iconList.length).toEqual(3)
    expect(iconList.first().props().icon).toEqual(itemsWithCategory[0].category.iconName)
  })
  it('should trigger the correct function callback', () => {
    const firstItem = wrapper.find('.list-group-item').first()
    firstItem.find('em').first().simulate('click')
    expect(props.onModifyItem).toHaveBeenCalledWith(itemsWithCategory[0])
    firstItem.find('em').last().simulate('click')
    expect(props.onDeleteItem).toHaveBeenCalledWith(itemsWithCategory[0])
  })
})