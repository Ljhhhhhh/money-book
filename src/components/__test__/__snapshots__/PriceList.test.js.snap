// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test pricelist component should render the compenent to match snapshot 1`] = `ShallowWrapper {}`;
